wallet/wallet.go

```diff
@@ -1320,6 +1342,17 @@ out:
 
 		// Select statement fell through by an explicit lock or the
 		// timer expiring.  Lock the manager here.
+
+		// We can't lock the manager if recovery is active because we use
+		// cryptoKeyPriv and cryptoKeyScript in recovery.
+		if recoverySyncI := w.recovering.Load(); recoverySyncI != nil {
+			recoverySync := recoverySyncI.(*recoverySyncer)
+			// If recovery is still running, it will end early with an error
+			// once we set the quit flag.
+			atomic.StoreUint32(&recoverySync.quit, 1)
+			<-recoverySync.done
```

Comment:
Should also select on the `quit` channel as well so we don't introduce the possibility of things deadlocking here.